core/src/test/java/org/apache/calcite/test/RelBuilderTest.java

```diff
@@ -2650,19 +2650,17 @@ private static RelBuilder assertSize(RelBuilder b,
         b.scan("EMP")
             .scan("DEPT")
             .join(JoinRelType.INNER,
-                b.or(b.literal(null),
-                    b.and(b.equals(b.field(2, 0, "DEPTNO"), b.literal(1)),
-                        b.equals(b.field(2, 0, "DEPTNO"), b.literal(2)),
-                        b.equals(b.field(2, 1, "DEPTNO"),
-                            b.field(2, 0, "DEPTNO")))))
+                b.and(b.equals(b.field(2, 0, "DEPTNO"), b.literal(1)),
```

Comment:
why are you modifying this test?

Reply:
Keeping a NULL literal makes the type of the condition to be of type NULL. This PR is not ready for review, and I am just testing things as tests are failing on my local machine with unrelated errors. I will mark this PR as draft for now.